packages/plugin-essentials/sources/commands/set/version/sources.ts

```diff
@@ -151,3 +98,60 @@ export default class SetVersionCommand extends BaseCommand {
     return report.exitCode();
   }
 }
+
+export async function runWorkflow(workflow: Array<Array<string>>, {configuration, context, target}: {configuration: Configuration, context: CommandContext, target: PortablePath}) {
+  for (const [fileName, ...args] of workflow) {
+    const usePipe = args[args.length - 1] === `|`;
+    if (usePipe)
+      args.pop();
+
+    if (usePipe) {
+      await execUtils.pipevp(fileName, args, {
+        cwd: target,
+        stdin: context.stdin,
+        stdout: context.stdout,
+        stderr: context.stderr,
+        strict: true,
+      });
+    } else {
+      context.stdout.write(`${configuration.format(`  $ ${[fileName, ...args].join(` `)}`, `grey`)}\n`);
+
+      try {
+        await execUtils.execvp(fileName, args, {
+          cwd: target,
+          strict: true,
+        });
+      } catch (error) {
+        context.stdout.write(error.stdout || error.stack);
+        throw error;
+      }
+    }
+  }
+}
+
+export async function prepareRepo<T extends Omit<SetVersionSourcesCommand, 'plugins'>>(this: T, {configuration, report, target}: {configuration: Configuration, report: Report, target: PortablePath}) {
```

Comment:
I prefer not to use `this` for free functions. What do you think of using a regular parameter, and typing it normally? Ie:

```ts
type PrepareSpec = {
  force: boolean;
};

function prepareRepo(spec: PrepareSpec, ...) {
  ...
}
```